datalad/core/local/configuration.py

```diff
@@ -0,0 +1,337 @@
+            else:
+                specs.append(tuple(s.split('=', 1)))
+
+        ds = None
+        if scope != 'global' or recursive:
+            ds = require_dataset(
+                dataset,
+                check_installed=True,
+                purpose='configuration')
+
+        res_kwargs = dict(
+            action='configuration',
+            logger=lgr,
+        )
+        if ds:
+            res_kwargs['refds'] = ds.path
+        yield from configuration(action, scope, specs, res_kwargs, ds)
+
+        if not recursive:
+            return
+
+        for subds in ds.subdatasets(
+                fulfilled=True,
+                recursive=True,
+                recursion_limit=recursion_limit,
+                on_failure='ignore',
+                result_renderer='disabled'):
+            yield from configuration(
+                action, scope, specs, res_kwargs, Dataset(subds['path']))
+
+    @staticmethod
+    def custom_result_renderer(res, **kwargs):
+        if (res['status'] != 'ok' or res['action'] != 'get_configuration'):
+            default_result_renderer(res)
+            return
+        # TODO source
+        from datalad.ui import ui
+        name = res['name']
+        if kwargs.get('action', None) == 'dump' and name in cfg_defs:
+            ui_def = cfg_defs[name].get('ui', [None, {}])[1]
+            for s in (ui_def.get('title'),
+                      ui_def.get('text')):
+                if s:
+                    ui.message('\n'.join(wrap(
+                        s,
+                        initial_indent='# ',
+                        subsequent_indent='# ',
+                    )))
+
+        if kwargs.get('recursive', False):
+            have_subds = res['path'] != res['refds']
+            # we need to mark up from which dataset results are reported
+            prefix = '<ds>{}{}:'.format(
+                '/' if have_subds else '',
+                Path(res['path']).relative_to(res['refds']).as_posix()
+                if have_subds else '',
+            )
+        else:
+            prefix = ''
+
+        if kwargs.get('action', None) == 'dump':
+            ui.message('{}{}={}'.format(
+                prefix,
+                ac.color_word(name, ac.BOLD),
+                res['value'] if res['value'] is not None else '',
+            ))
+        else:
+            ui.message('{}{}'.format(
+                prefix,
+                res['value'] if res['value'] is not None else '',
+            ))
+
+
+def configuration(action, scope, specs, res_kwargs, ds=None):
+    if scope == 'global':
+        cfg = dlcfg
+    else:
+        cfg = ds.config
+
+    if action not in config_actions:
+        raise ValueError("Unsupported action '{}'".format(action))
+
+    if action == 'dump':
+        if specs:
+            lgr.warning('Provided configuration selection is ignored when dumping')
+        # dumping is querying for all known keys
+        # TODO warn that specs will be ignored, if there are any
```

Comment:
It looks like this todo is covered by the warning above.  However, I think the warning will be very easy to miss with the verbose output of `dump`.  I'd vote for changing this warning to a value error early in `__call__`.

Also, should a similar warning/error also happen with `--scope`?

Reply:
Yes, you are right. Done.